frontend/packages/kubevirt-plugin/src/components/modals/delete-device-modal/delete-device-modal.tsx

```diff
@@ -57,10 +55,8 @@ export const DeleteDeviceModal = withHandlePromise((props: DeleteDeviceModalProp
     if (!patches || patches.length === 0) {
       close();
     } else {
-      const model = isVm(vmLikeEntity) ? VirtualMachineModel : TemplateModel;
-      const promise = k8sPatch(model, vmLikeEntity, patches);
-      // eslint-disable-next-line promise/catch-or-return
-      handlePromise(promise).then(close);
+      const promise = k8sPatch(getVmLikeModel(vmLikeEntity), vmLikeEntity, patches);
+      handlePromise(promise).then(close); // eslint-disable-line promise/catch-or-return
```

Comment:
Shouldn't we handle promise rejection? Why are we suppressing the [`promise/catch-or-return`](https://github.com/xjamundx/eslint-plugin-promise/blob/master/docs/rules/catch-or-return.md) rule here?

Core `coFetch` function uses the [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch) API along with a timeout promise, so in practice, `k8sPatch` can fail on either network error or request timeout.

Reply:
Rejection is handled by `handlePromise` and the error is shown in that case.

I could not make it work unless I do:
```ts
if (patches.length === 0) {
  close();
  return Promise.resolve(null);
}
const promise = k8sPatch(getVMLikeModel(vmLikeEntity), vmLikeEntity, patches);
return handlePromise(promise).then(close);
```

which seems a bit over the top to me for a function which is passed to `onSubmit`

note:
normal return shows lint error: `error  Arrow function expected no return value  consistent-return`

Reply:
> Rejection is handled by `handlePromise` and the error is shown in that case.

You are right, the `withHandlePromise` HOC handles the error as well. I've missed that.

I'm fine with `// eslint-disable-line promise/catch-or-return` here.